security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -251,18 +253,26 @@ func (sc *SecretCache) GenerateSecret(ctx context.Context, connectionID, resourc
 	// the files under the well known path.
 	sdsFromFile := false
 	var err error
+
 	if connKey.ResourceName == RootCertReqResourceName && sc.rootCertificateExist(sc.existingRootCertFile) {
 		sdsFromFile = true
 		ns, err = sc.generateRootCertFromExistingFile(sc.existingRootCertFile, token, connKey)
-	} else if connKey.ResourceName == WorkloadKeyCertResourceName &&
-		sc.keyCertificateExist(sc.existingCertChainFile, sc.existingKeyFile) {
+	} else if cfg, ok := pilotmodel.SdsCertificateConfigFromResourceName(connKey.ResourceName); ok && cfg.CaCertificatePath != "" {
```

Comment:
Nit: I would add some comment for these conditions. I need to read carefully to see the difference between CaCertificatePath and CertificatePath.

Reply:
Sure. Will add in follow-up